src/styles/main.scss

```diff
@@ -1,3 +1,55 @@
+@import './variables';
+
 body {
   margin: 0;
+  font-family: Arial, sans-serif;
+  display: flex;
+  align-items: center;
+  justify-content: center;
+  height: 100vh;
+}
+
+.calendar {
+  display: flex;
+  flex-wrap: wrap;
+  padding: 10px;
+  gap: $gap;
+  width: $day-rows * $box-size + ($day-rows - 1) * $gap;
+
+  @each $day, $day-number in $days {
+    &--start-day-#{$day} &__day:first-child {
+      margin-left: $day-number * ($box-size + $gap);
+    }
+  }
+
+  @for $i from 28 through 31 {
+    &--month-length-#{$i} &__day:nth-child(n + #{$i + 1}) {
+      display: none;
+    }
+  }
+
+  &__day {
+    display: flex;
+    justify-content: center;
+    align-items: center;
+    width: $box-size;
+    height: $box-size;
+    box-sizing: border-box;
+    font-size: 30px;
+    background-color: $box-color;
+    border: solid 1px $box-border-color;
+    transition: all 0.5s;
```

Comment:
It is recommended to specify the properties instead of 'all' for better performance